spec/lib/signed-token-request.spec.js

```diff
@@ -0,0 +1,253 @@
+"use strict";
+
+describe("SignedTokenRequest", () => {
+    var bluebird, container, fsMock, privateSigned, publicSigned, requestAsyncMock, requestHandler, requestMock, responseMock, SignedTokenRequest, signer, StreamReadable, url;
+
+    // Just some test data that is constant.
+    const account = {
+            code: "code",
+            secret: "secret",
+            id: "accountId"
+        },
+        host = "api.opentoken.io",
+        token = "token";
+
+    beforeEach(() => {
+        container = require("../../lib/container")(host, "info");
+        bluebird = container.resolve("bluebird");
+        requestMock = require("../mock/request-mock")();
+        requestMock.get.andReturn(requestMock);
+        requestMock.post.andReturn(requestMock);
+        container.register("request", requestMock);
+        responseMock = require("../mock/response-mock")();
+        container.register("logger", require("../mock/logger-mock")());
+        fsMock = jasmine.createSpyObj("fs", ["createWriteStream"]);
+        container.register("fs", fsMock);
+        requestAsyncMock = jasmine.createSpy("requestAsync");
+        container.register("requestAsync", requestAsyncMock);
+        signer = container.resolve("signer");
+        spyOn(signer, "sign").andCallThrough();
+        requestHandler = container.resolve("requestHandler");
+        spyOn(requestHandler, "createUrl").andCallThrough();
+        spyOn(requestHandler, "checkResponse").andCallThrough();
+        spyOn(requestHandler, "buildRequestOptions").andCallThrough();
+        SignedTokenRequest = container.resolve("SignedTokenRequest");
+        StreamReadable = container.resolve("StreamReadable");
+        url = `https://${host}/account/${account.id}/token`;
+        privateSigned = new SignedTokenRequest(account.id, false, account.code, account.secret);
+        publicSigned = new SignedTokenRequest(account.id, true, account.code, account.secret);
```

Comment:
They moved down like two lines. They cannot move much more then that. The get.andReturn is used by multiple tests so it is in the beforeEach.